src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/todo';
+import { useState } from 'react';
+
+type TodosProperties = {
+  todos: Todo[];
+  titleValue: string;
+  userValue: string;
+  isFormSubmited: boolean;
+};
+
+type HandleChangeTodosState =
+  | React.ChangeEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLSelectElement>;
+
+const todosProperties: TodosProperties = {
+  todos: todosFromServer,
+  titleValue: '',
+  userValue: '0',
+  isFormSubmited: false,
+};
 
 export const App = () => {
+  const [todosState, setTodosState] =
+    useState<TodosProperties>(todosProperties);
+  const { todos, titleValue, userValue, isFormSubmited } = todosState;
+
+  const handleSubmitForm = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!titleValue.length || userValue === '0') {
+      return setTodosState(prev => ({
+        ...prev,
+        isFormSubmited: true,
+      }));
+    }
+
+    const userIdSearch = usersFromServer.find(
+      user => user.name === userValue,
+    )?.id;
+
+    const newTodo: Todo = {
+      id: todos.length + 1,
```

Comment:
instead of creating id based on length you should find max id and increase it:
![image](https://github.com/user-attachments/assets/4dd682ad-00ec-47ae-a1c1-7f8e6cf6638a)